cache/core.go

```diff
@@ -4,51 +4,44 @@ import (
 	"database/sql"
 	"time"
 
-	log "github.com/Sirupsen/logrus"
-
 	"github.com/Bnei-Baruch/archive-backend/utils"
 )
 
-type Refreshable interface {
-	Refresh() error
-}
-
-type Provider interface {
-	Refreshable
-	String() string
-}
-
 type CacheManager interface {
 	SearchStats() SearchStatsCache
+	SourcesStats() SourcesStatsCache
+	TagsStats() TagsStatsCache
 	Close()
 }
 
 type CacheManagerImpl struct {
-	ssc              SearchStatsCache
+	search           SearchStatsCache
+	sources          SourcesStatsCache
+	tags             TagsStatsCache
 	ticker           *time.Ticker
 	ticks            int64
 	refreshIntervals map[string]int64
 }
 
 func NewCacheManagerImpl(mdb *sql.DB, refreshIntervals map[string]time.Duration) CacheManager {
 	cm := new(CacheManagerImpl)
-	cm.ssc = NewSearchStatsCacheImpl(mdb)
-	cm.refresh(cm.ssc)
-
 	// Convert time.Duration to int64
 	// So we would have refresh intervals in integer multiple of a second
 	cm.refreshIntervals = make(map[string]int64, len(refreshIntervals))
 	for k, v := range refreshIntervals {
 		cm.refreshIntervals[k] = int64(v.Truncate(time.Second).Seconds())
 	}
 
+	cm.sources = NewSourcesStatsCacheImpl(mdb)
+	cm.tags = NewTagsStatsCacheImpl(mdb)
+	cm.search = NewSearchStatsCacheImpl(mdb)
+	cm.refresh()
+
 	cm.ticker = time.NewTicker(time.Second)
 	go func() {
 		for range cm.ticker.C {
 			cm.ticks++
-			if cm.ticks%cm.refreshIntervals["SearchStats"] == 0 {
```

Comment:
This should not change once `Refresh` above is not changed

Reply:
changed